src/main/native/GCM.c

```diff
@@ -1539,7 +1539,7 @@ JNIEXPORT jint JNICALL Java_com_ibm_crypto_plus_provider_ock_NativeInterface_do_
 
 #ifdef DEBUG_GCM_DETAIL
     if ( debug ) {
-        gslogMessage ("NI do_GCM_UpdForUpdateEncrypt cls %x\n", cls);
+        gslogMessage ("NI do_GCM_UpdForUpdateEncrypt thisObj %x\n", thisObj);
```

Comment:
In this case the JNI passes the class, not the object I think. So, the name is a bit confusing. In other JNI code we specify the parameter as `jclass clazz`.

Reply:
Looking into this some there are various documents including [this one ](https://docs.oracle.com/en/java/javase/22/docs/specs/jni/design.html) that state something like the following:

---

_The second argument differs depending on whether the native method is static or non-static. The second argument to a non-static native method is a reference to the object. The second argument to a static native method is a reference to its Java class._

---

So I believe that you are right here that the reference is really a reference to a Java class not the object itself given that all our methods in `NativeInterface.java` are all declared as static. 

For consistancy reasons it seems we could just leave this as `thisObj` given that almost all the JNI code in the project uses `thisObj` instead of `clazz`. If we want to make the change for this variable name perhaps we could do this in another PR that changes all variables from `thisObj` to `clazz`.

Reply:
Sure. We could open another PR to fix those kind of things.